### PR TITLE
fix: target and container size initialization order should not affect base transform calculations

### PR DESCRIPTION
**Pull request by tpoisseau**
https://repro-scale-issue.react-roi.pages.dev/?path=%2Fstory%2Fedge-cases-zoom--no-config-without-target-image

https://repro-scale-issue.react-roi.pages.dev/?path=%2Fstory%2Fedge-cases-zoom--initial-config-zoom-without-target-image

### CODE REVIEW

**Comment by stropitek** [comment]
@tpoisseau there was an issue with the initialization of the base transform (the one which fits the target relative to the container, according to `resizeStrategy`). It is calculated once both the container and target sizes are known, but the order was affecting the result.

Note that the initial placement of the target can be affected by a few parameters, not only the initial zoom level:
- `resizeStrategy`
- `zoom.spaceAroundTarget`: this can "correct" the zoom transform when the placement of the target is is too off-center. If you test the edge cases stories you added with the `resizeStrategy: "none"`, you'll see a strange result because of its default value of 0.5.

**Comment by stropitek** [comment]
I let you review and ping me since you cannot approve.

File: src/context/RoiProvider.tsx [hunk 1]
```diff
@@ -71,6 +71,17 @@ export interface RoiProviderInitialConfig<TData> {
...
+   * Defines how the target should fit within the container at initialization.
+   * - `none`: The target is not transformed relative to the container. The top-left corner of the target matches the
+   *           top-left corner of the container.
+   * - `contain`: The target is transformed such as the full target is visible in the container. The center of the
+   *              target matches the center of the container.
+   * - `cover`: The target is scaled down such that one of the dimensions is fitting the container exactly, or not
+   *            scaled at all if the target is smaller than the container. The center of the target matches the center
+   *            of the container.
+   * - `center`: The center of the target matches the center of the container, without scaling the target.
+   */
```
Comment (tpoisseau) [inline src/context/RoiProvider.tsx:85]: Is there a default value? If not, what is the behavior of `undefined`?

**Review by tpoisseau — COMMENTED** [review]
Otherwise it looks good to me, I used yalc to check on the project where I discover the issue and it's OK now.

@stropitek You can merge (after you comment `undefined` behavior for `resizeStrategy`)

File: src/context/RoiProvider.tsx (hunk 1 above)
Comment (stropitek) [reply src/context/RoiProvider.tsx:85]: The default value is `contain`

**Review by stropitek — APPROVED** [review]
